src/liballoc/task.rs

```diff
@@ -18,24 +18,75 @@ use crate::sync::Arc;
 /// exists as an alternative for those systems.
 ///
 /// [arc]: ../../std/sync/struct.Arc.html
-#[unstable(feature = "wake_trait", issue = "69912")]
+///
+/// # Examples
+/// ```rust
+/// use std::future::Future;
+/// use std::sync::Arc;
+/// use std::task::{Context, Poll, Wake};
+/// use std::thread::{self, Thread};
+/// use std::pin::Pin;
+/// 
+/// /// A waker that wakes up the current thread when woken.
+/// struct ThreadWaker(Thread);
+/// 
+/// impl Wake for ThreadWaker {
+///     fn wake(self: Arc<Self>) {
+///         self.0.unpark();
+///     }
+/// }
+/// 
+/// /// Run a future to completion on the current thread.
+/// ///
+/// /// # Safety
+/// ///
+/// /// This method may deadlock in nested calls; this should
+/// /// only be called once at the root of fn main.
+/// fn block_on<F, T>(mut fut: F) -> T
+/// where
+///     F: Future<Output = T>,
+/// {
+///     // SAFETY: The future is not moved or mutated after this.
```

Comment:
I'm unsure about the phrasing of this comment. In ecosystem code we would use `pin_mut!` at this site, but we don't have access to it in the stdlib.

cc/ @taiki-e any suggestions on the phrasing here?

Reply:
Could this just use `Box::pin`? That would allow this example to be completely safe.

Reply:
Oh yeah, good suggestion — didn't consider that. Will change!